library/core/src/option.rs

```diff
@@ -1049,6 +1049,47 @@ impl<T> Option<T> {
     {
         Some(f(self?, other?))
     }
+
+    /// Zips `self` and another `Option` with function `f`, or returns `self.or(other)`.
+    ///
+    /// If `self` is `Some(s)` and `other` is `Some(o)`, this method returns `Some(f(s, o))`.
+    /// Otherwise, if `self` is `Some`, `self` is returned.
+    /// Otherwise, `other` is returned.
```

Comment:
```suggestion
    /// If `self` is `Some(s)` and `other` is `Some(o)`, this method zips the two values
    /// with the provided function, returning `Some(f(s, o))`.
    /// Otherwise, the result of [`self.or(other)`] is returned.
```

Reply:
I've made this change (accounting for the name change).